src/ngraph/node.hpp

```diff
@@ -193,6 +193,9 @@ namespace ngraph
         virtual const op::AutoBroadcastSpec& get_autob() const;
         /// \returns true if the node can decompose
         virtual bool supports_decompose() const { return false; }
+        /// \brief Can replace a node with a constant during constant folding.
+        /// \returns vector of outputs to replace node outputs. Empty outputs will not be replace.
+        virtual OutputVector constant_fold_default() { return {}; }
```

Comment:
Please rename to `constant_fold`, without `_default`. I don't know what `default` means in this case. This is _only_ place where we define constant folding for an operation. We are going to migrate all code for constant folding under this API. With this renaming the rest of the code looks good to me.